Coordination Implementation

EDOLK opened the pull request:
Coordinates the program mainly through the new mediator object. User input comes from the command line to CommandLineEntryPoint object, we should be able to implement the UserRequestProvider interface for different types of input (http requests, REST, etc). This probably won't pass checks because it references things in both of your prs, we'll have to re-run checks after the other prs are merged.

EDOLK committed:
adds coordination implementation

kyliedelamater reviewed (approved):
everything looks good! the only thing I have to say is the Mediator class sendInput method has a lot of stuff in it and it's a bit hard to follow. i'm also not completely sure what the Mediator class is doing, but i'm assuming it makes the process of sending the file sources easier.

EDOLK committed:
synch with main (#27)

* ComputeEngineImplementation: Added Implementation for compute engine computation

* Updated computation implementation

* assignment #4 part 3 and other fixes (#23)

* updated DataStoreAPI with read/write configs

fileReader and Scanner used for reading the input and fileWriter used for writing the output. i added quite a bit more to make tests run as well.

* Update ComputeEngineAPI.java

* Update InputRequest.java

* Update InputResponse.java

* Update OutputRequest.java

* Update OutputResponse.java

* Update TestDataStorage.java

* Update InputInMemoryImplementation.java

* Update OutputInMemoryImplementation.java

* Update DataStoreInMemoryImplementation.java

* Update DataStoreInMemoryImplementation.java

* Update DataStoreInMemoryImplementation.java

---------

Co-authored-by: Fv40 <deckerd70@yahoo.com>
Co-authored-by: WaveyD <76503134+deckerd70@users.noreply.github.com>
Co-authored-by: Fv40 <151473283+Fv40@users.noreply.github.com>
Co-authored-by: kyliedelamater <kyliedelamater@gmail.com>

EDOLK committed:
fixed mediator
